nncf/internal/deprecation.py

```diff
@@ -0,0 +1,42 @@
+""""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import warnings
+from typing import Callable
+
+
+def warning_deprecated(msg):
+    # Note: must use FutureWarning in order not to get suppressed by default
+    warnings.warn(msg, FutureWarning, stacklevel=2)
+
+
+class deprecated_call:
```

Comment:
why it is not just `deprecated`? It can be applicable to classes and methods.

Reply:
Will try to come up with an implementation that works for class instantiation as well.

Reply:
Done, but the resulting decorator will only trigger a warning on instantiation of the class objects. Enums and static attributes of classes cannot be deprecated in this fashion.